integrations/GA/browser.js

```diff
@@ -1,122 +1,986 @@
+        var products = properties.products;
+
+        if (!orderId) return;
+
+        if (!this.ecommerce) {
+          ga("require", "ecommerce");
+          this.ecommerce = true;
+        }
+
+        ga("ecommerce:addTransaction", {
+          affiliation: properties.affiliation,
+          shipping: properties.shipping,
+          revenue: total,
+          tax: properties.tax,
+          id: orderId,
+          currency: properties.currency,
+        });
+
+        each(products, function (product) {
+          var productTrack = createProductTrack(rudderElement, product);
+
+          ga("ecommerce:addItem", {
+            category: productTrack.category,
+            quantity: productTrack.quantity,
+            price: productTrack.price,
+            name: productTrack.name,
+            sku: productTrack.sku,
+            id: orderId,
+            currency: productTrack.currency,
+          });
+        });
+
+        ga("ecommerce:send");
+      }
+      // enhanced ecommerce events
+      else if (
+        (event === "Checkout Started" ||
+          event === "Checkout Step Viewed" ||
+          event === "Order Updated") &&
+        this.enhancedEcommerce
+      ) {
+        var self = this;
 
-    var payLoad = {
-      hitType: "event",
-      eventCategory: eventCategory,
-      eventAction: eventAction,
-      eventLabel: eventLabel,
-      eventValue: eventValue
-    };
-    ga("rudder_ga.send", "event", payLoad);
-    logger.debug("in GoogleAnalyticsManager track");
+        var properties = rudderElement.message.properties;
+        var products = properties.products;
+        var options = extractCheckoutOptions(rudderElement);
+
+        this.enhancedEcommerceLoaded = loadEnhancedEcommerce(
+          rudderElement,
+          this.enhancedEcommerceLoaded
+        );
+        each(products, function (product) {
+          var productTrack = createProductTrack(rudderElement, product);
+          productTrack = { message: productTrack };
+
+          enhancedEcommerceTrackProduct(productTrack, self.inputs);
+        });
+
+        ga("ec:setAction", "checkout", {
+          step: properties.step || 1,
+          option: options || undefined,
+        });
+
+        pushEnhancedEcommerce(rudderElement, this.inputs);
+      } else if (
+        event === "Checkout Step Completed" &&
+        this.enhancedEcommerce
+      ) {
+        var props = rudderElement.message.properties;
+        var options = extractCheckoutOptions(rudderElement);
+
+        if (!props.step) return;
+
+        var params = {
+          step: props.step || 1,
+          option: options || undefined,
+        };
+
+        this.enhancedEcommerceLoaded = loadEnhancedEcommerce(
+          rudderElement,
+          this.enhancedEcommerceLoaded
+        );
+
+        ga("ec:setAction", "checkout_option", params);
+        ga("send", "event", "Checkout", "Option");
+      } else if (event === "Order Completed" && this.enhancedEcommerce) {
```

Comment:
add the this.enhancedEcommerce check at the top level condition then switch-case?

Reply:
done